### ISSUE TITLE
Need JavaScript Proxy support on Android (upgrade JSCore)

### ISSUE THREAD

**Issue by rclai** [expo/expo#1874]
Mobx has just released version 5 which uses the Proxy object under the hood, but out of the box, Android doesn't have support for that
https://github.com/mobxjs/mobx#browser-support

So we have to upgrade JSCore manually but this is not doable in Expo
https://github.com/react-community/jsc-android-buildscripts#how-to-use-it-with-my-react-native-app

I would post this on [Canny](https://expo.canny.io/feature-requests/p/javascript-proxy-support-for-android-upgrade-jscore) except it would take forever for it gain enough votes to even make it to your eyes in addition to how much less love Android gets in general.

Essentially anyone using Expo right now would be stuck in a sub version 5 of Mobx if this is not done.

Related issue: https://github.com/mobxjs/mobx/issues/1587

**Comment by brentvatne** [comment]
we updated a couple times in the past but there were bugs that forced us to revert the update. I think we will revisit this soon.

**Comment by brentvatne** [comment]
also, have you tried using a polyfill?

**Comment by rclai** [comment]
The polyfill is not ideal because of the issues mentioned on here:
https://stackoverflow.com/questions/35025204/javascript-proxy-support-in-babel

**TLDR**
- babel-proxy-plugin
   - performance impact
   - author doesn't recommend for production
- google proxy-polyfill
    - limited suport
    - dynamic object fields not supported, which doesn't play well with Mobx

**Comment by rclai** [comment]
Great to hear!

**Comment by mekwall** [comment]
@brentvatne Since this didn't make it into SDK 29, when are we expected to see it? And please, update the docs at least. We upgraded to MobX 5 because the docs say that both Symbol and Proxy are supported by JSC (on Android and iOS 10+) so we were in the clear.

**Comment by skenan** [comment]
Any update?

**Comment by terrysahaidak** [comment]
Does the Expo SDK 30 include updated JSCore?

**Comment by meepeek** [comment]
I am stuck in this issue as well.

**Comment by skenan** [comment]
@sjchmiela  It solved the issue after upgrading to 31. Thank you!

**Comment by sjchmiela** [comment]
Glad to hear that! 🎉

**Comment by rclai** [comment]
Yay!
